### PR TITLE
AP_Math: header and const cleanup

### PR DESCRIPTION
**Pull request by khancyr**
This PR intend to include the right header in the right place.
It unifies the header include by using c++ version instead of mixing both C and C++ one's
Solve an access error on a private variable
Const local variable
Correct some scope and assignement

Size is unchanged on fmuv3, on SITL debug f26b60a take a little more space

### CODE REVIEW

**Comment by khancyr** [comment]
those were dectected by cppcheck, clang-tidy

File: libraries/AP_Math/quaternion.h [hunk 1]
```diff
@@ -21,7 +21,10 @@
 #if MATH_CHECK_INDEXES
 #include <assert.h>
 #endif
-#include <math.h>
+#include <math.h>  // needed for isnan
```
Comment (OXINARF) [inline libraries/AP_Math/quaternion.h]: Please explain, build system should take care of fixing this problem.

Comment (khancyr) [reply libraries/AP_Math/quaternion.h]: I don't know how I did get an error on that, but it seems working now !

**Comment by khancyr** [comment]
rebased on master

**Comment by khancyr** [comment]
rebased on master
build on master
````
BUILD SUMMARY
Build directory: /home/pierre/Workspace/ardupilot/build/sitl
Target               Text     Data   BSS    Total  
---------------------------------------------------
bin/ardurover        1888830  72438  42072  2003340
bin/antennatracker   1430887  50776  37272  1518935
bin/arducopter       2246214  82686  47512  2376412
bin/arducopter-heli  2209958  85374  47384  2342716
bin/arduplane        2194942  87662  46136  2328740
bin/ardusub          1848470  65142  40312  1953924

````

build on PR 
````
BUILD SUMMARY
Build directory: /home/pierre/Workspace/ardupilot/build/sitl
Target               Text     Data   BSS    Total  
---------------------------------------------------
bin/ardurover        1888156  72446  42072  2002674
bin/antennatracker   1430213  50784  37272  1518269
bin/arducopter       2245540  82694  47512  2375746
bin/arducopter-heli  2209268  85382  47384  2342034
bin/arduplane        2194252  87670  46136  2328058
bin/ardusub          1847796  65150  40312  1953258

````

**Comment by magicrub** [comment]
PR of the year award?

**Comment by khancyr** [comment]
at least it is something ...

**Comment by magicrub** [comment]
@khancyr I'm not being sarcastic. Boring infrastructure improvements like this usually go without celebration. It is very important that we do the boring infrasture work so everything else works so well. Thank you!

**Comment by rmackay9** [comment]
I've merged in two of the commits.  I don't normally merge in only part of a PR but I worry the PR will get stuck so I did what I could. Maybe rebase it and add it as a dev call topic.

**Comment by khancyr** [comment]
After review with @peterbarker, I close this one in favor of https://github.com/ArduPilot/ardupilot/pull/10426/

**Pull request closed without merging**